proxy/http/HttpTransactCache.cc

```diff
@@ -30,126 +30,84 @@
 #include "HTTP.h"
 #include "HttpCompat.h"
 #include "tscore/InkErrno.h"
+#include "tscpp/util/TextView.h"
 
-/**
-  Find the pointer and length of an etag, after stripping off any leading
-  "W/" prefix, and surrounding double quotes.
+using ts::TextView;
 
-*/
-inline static const char *
-find_etag(const char *raw_tag_field, int raw_tag_field_len, int *length)
+namespace
 {
-  const char *quote;
-  int etag_length        = 0;
-  const char *etag_start = raw_tag_field;
-  const char *etag_end   = raw_tag_field + raw_tag_field_len;
-
-  if ((raw_tag_field_len >= 2) && (etag_start[0] == 'W' && etag_start[1] == '/')) {
-    etag_start += 2;
-  }
-
-  etag_length = etag_end - etag_start;
-
-  if ((etag_start < etag_end) && (*etag_start == '"')) {
-    ++etag_start;
-    --etag_length;
-    quote = (const char *)memchr(etag_start, '"', etag_length);
-    if (quote) {
-      etag_length = quote - etag_start;
-    }
-  }
-  *length = etag_length;
-  return etag_start;
-}
+constexpr TextView ETAG_WEAK_MATCH_PREFIX{"W/"};
 
 /**
-  Match an etag raw_tag_field with a list of tags in the comma-separated
-  string field_to_match, using strong rules.
+  Find the pointer and length of an etag, after stripping off any leading
+  "W/" prefix, and surrounding double quotes.
 
 */
-inline static bool
-do_strings_match_strongly(const char *raw_tag_field, int raw_tag_field_len, const char *comma_sep_tag_list,
-                          int comma_sep_tag_list_len)
+TextView
+extract_etag(TextView field)
 {
-  StrList tag_list;
-  const char *etag_start;
-  int n, etag_length;
-
-  // Can never match a weak tag with a strong compare
-  if ((raw_tag_field_len >= 2) && (raw_tag_field[0] == 'W' && raw_tag_field[1] == '/')) {
-    return false;
-  }
-  // Find the unalterated tag
-  etag_start = find_etag(raw_tag_field, raw_tag_field_len, &etag_length);
-
-  // Rip the field list into a comma-separated field list
-  HttpCompat::parse_comma_list(&tag_list, comma_sep_tag_list, comma_sep_tag_list_len);
-
-  // Loop over all the tags in the tag list
-  for (Str *tag = tag_list.head; tag; tag = tag->next) {
-    // If field is "*", then we got a match
-    if ((tag->len == 1) && (tag->str[0] == '*')) {
-      return true;
+  if (!field.empty()) {
```

Comment:
The elimination of `StrList` is the primary performance win for this change. `StrList` contains a linked list of token objects, which is allocated in `HttpCompat::parse_comma_list`. Switching to `TextView` means the tokens can be extracted as a stream without modifying the original string (which, because it's a value in the header, is not permitted). Therefore this logic goes from an allocation per ETAG found to none.